hazelcast-client/src/main/java/com/hazelcast/client/config/XmlClientConfigBuilder.java

```diff
@@ -92,6 +100,38 @@ public XmlClientConfigBuilder() {
         this.in = locator.getIn();
     }
 
+    @Override
+    protected Document parse(InputStream inputStream) throws Exception {
+
+        final DocumentBuilder builder = DocumentBuilderFactory.newInstance().newDocumentBuilder();
+        try {
+            return builder.parse(inputStream);
+        } catch (final Exception e) {
+            String lineSeparator = StringUtil.getLineSeperator();
+            String msg = "Failed to parse Config Stream"
+                    + lineSeparator + "Exception: " + e.getMessage()
+                    + lineSeparator + "HazelcastClient startup interrupted.";
+            LOGGER.severe(msg);
+            throw e;
+        } finally {
+            IOUtil.closeResource(inputStream);
+        }
+    }
+
+    @Override
+    protected Properties getProperties() {
+        return properties;
+    }
+
+    public void setProperties(Properties properties) {
```

Comment:
if user wants to pass properties object other than setting system properties.
like here: https://github.com/hazelcast/hazelcast/pull/4581/files#diff-eaa5dbc388de7caa71a251009b4e8d6cR232
In server side we support this.